sklearn/metrics/tests/test_ranking.py

```diff
@@ -610,6 +610,17 @@ def test_multiclass_ovr_roc_auc_toydata(y_true, labels):
         result_weighted,
     )
 
+    # Perfect classifier has roc_auc_score = 1.0
+    y_perfect = [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]
+    assert_almost_equal(
+        roc_auc_score(y_true, y_perfect, multi_class="ovr", labels=labels),
+        1.0,
+    )
+
+    # Imperfect classifier has roc_auc_score < 1.0
+    y_imperfect = [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0], [1.0, 0.0, 0.0]]
+    assert roc_auc_score(y_true, y_imperfect, multi_class="ovr", labels=labels) < 1.0
```

Comment:
Could you please parameterize those two test cases (perfect AUC and non-perfect AUC) for all supported averaging methods? I think we can move them in a dedicated test function since the beginning of the test function contains assertions that are specific to a particular choice of averaging.